src/plugins/discover/public/application/main/hooks/utils/build_state_subscribe.ts

```diff
@@ -47,15 +47,29 @@ export const buildStateSubscribe =
     const nextQuery = nextState.query;
     const savedSearch = savedSearchState.getState();
     const prevQuery = savedSearch.searchSource.getField('query');
+    const isTextBasedQueryLang = isTextBasedQuery(nextQuery);
     const queryChanged = !isEqual(nextQuery, prevQuery) || !isEqual(nextQuery, prevState.query);
+    // check if prevState and nextState are equal except for index
+
+    if (
+      isTextBasedQueryLang &&
+      isEqualState(prevState, nextState, ['index', 'viewMode']) &&
+      !queryChanged
```

Comment:
Is the intent here to skip the rest of the function when `index`, `viewMode` and query haven't changed?